### PR TITLE
Write unit tests for getAllMenteesByProgramId

### LINKED ISSUE #128: Write a unit test for getAllMenteesByProgramId method
**Is your feature request related to a problem? Please describe.**
- We need to write unit tests getAllMenteesByProgramId method.

**Describe the solution you'd like**
- Write the relevant unit tests

### PR DESCRIPTION
**Pull request by posabogdanpetre**
## Purpose

The purpose of this PR is to fix #128 

## Goals

Write unit tests for getAllMenteesByProgramId

## Approach

Add unit tests

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Test environment
 
N/A

## Learning

N/A

### CODE REVIEW

File: src/test/java/org/sefglobal/scholarx/service/ProgramServiceTest.java [hunk 1]
```diff
@@ -326,4 +326,33 @@ void getSelectedMentorOfMentee_withUnavailableData_thenThrowNoContent() {
                 .isInstanceOf(NoContentException.class)
                 .hasMessage("Error, Mentee is not approved by any mentor yet.");
     }
+
+    @Test
+    void getAllMenteesByProgramIde_withUnavailableProgramId_thenThrowResourceNotFoundException() {
```
Comment (Piumal1999) [inline src/test/java/org/sefglobal/scholarx/service/ProgramServiceTest.java:331]:
```suggestion
    void getAllMenteesByProgramId_withUnavailableProgramId_thenThrowResourceNotFoundException() {
```

File: src/test/java/org/sefglobal/scholarx/service/ProgramServiceTest.java [hunk 2]
```diff
@@ -326,4 +326,33 @@ void getSelectedMentorOfMentee_withUnavailableData_thenThrowNoContent() {
...
+
+        Throwable thrown = catchThrowable(
+                () -> programService.getAllMenteesByProgramId(programId));
+        assertThat(thrown)
+                .isInstanceOf(ResourceNotFoundException.class)
+                .hasMessage("Error, Program by id: 9 doesn't exist");
+    }
+
+    @Test
+    void getAllMenteesByProgramIde_withAvailableData_thenCorrectDataFromRepositoryIsReturned() throws ResourceNotFoundException {
```
Comment (Piumal1999) [inline src/test/java/org/sefglobal/scholarx/service/ProgramServiceTest.java:345]:
```suggestion
    void getAllMenteesByProgramId_withAvailableData_thenReturnDataFromRepository() throws ResourceNotFoundException {
```

File: src/test/java/org/sefglobal/scholarx/service/ProgramServiceTest.java [hunk 3]
```diff
@@ -326,4 +326,33 @@ void getSelectedMentorOfMentee_withUnavailableData_thenThrowNoContent() {
...
+        Long programId = 9L;
+        doReturn(true)
+                .when(programRepository)
+                .existsById(programId);
+        List<Mentee> storedData = new ArrayList<>();
+        doReturn(storedData)
+                .when(menteeRepository)
+                .findAllByProgramId(programId);
+
+        List<Mentee> returnedDate = programService.getAllMenteesByProgramId(programId);
```
Comment (Piumal1999) [inline src/test/java/org/sefglobal/scholarx/service/ProgramServiceTest.java:355]:
```suggestion
        List<Mentee> returnedData = programService.getAllMenteesByProgramId(programId);
```

**Review by Piumal1999 — CHANGES_REQUESTED** [review]
Awesome! @posabogdanpetre 

There were some spelling mistakes. Could you please correct those?

File: src/test/java/org/sefglobal/scholarx/service/ProgramServiceTest.java (hunk 1 above)
Comment (posabogdanpetre) [reply src/test/java/org/sefglobal/scholarx/service/ProgramServiceTest.java]: Ups...thanks @Piumal1999!
Fixed them.

File: src/test/java/org/sefglobal/scholarx/service/ProgramServiceTest.java (hunk 3 above)
Comment (posabogdanpetre) [reply src/test/java/org/sefglobal/scholarx/service/ProgramServiceTest.java]: Fixed, thanks!

**Review by Piumal1999 — APPROVED** [review]
Well done! Thanks @posabogdanpetre

**Pull request merged**